caddyconfig/httploader.go

```diff
@@ -134,16 +134,16 @@ func doHttpCallWithRetries(ctx caddy.Context, client *http.Client, request *http
 	var err error
 	const maxAttempts = 10
 
-	// attempt up to 10 times
 	for i := 0; i < maxAttempts; i++ {
 		resp, err = attemptHttpCall(client, request)
 		if err != nil && i < maxAttempts-1 {
-			// wait 500ms before reattempting, or until context is done
 			select {
 			case <-time.After(time.Millisecond * 500):
 			case <-ctx.Done():
 				return resp, ctx.Err()
 			}
+		} else {
```

Comment:
I am a big advocate of return early and the succeeding patterns that follow from it, but in this case it made the code less readable.

Reply:
:thinking: I can't see how it would be less readable. In english, I'd read it like this:

> If there was an error in the attempt, or we reached the maximum amount of attempts, break (else continue with the select logic)

Reply:
It's either continue in the error case, break in the succesful case, or return early in the succesful case. The return would have been the same as the return on the end of the function, so I wanted to avoid that duplication and chose for break. Feel free to update it if you want to see it differently.

Reply:
I stubbed out the alternative and I agree that this is an odd case where this flows a little better IMO. It's purely preference but I think we can leave it (unless you feel strongly enough, feel free to change it).